src/OrleansTestingHost/TestCluster.cs

```diff
@@ -107,7 +110,7 @@ public void Deploy(IEnumerable<string> siloNames)
         /// Deploys the cluster using the specified configuration and starts the client in-process.
         /// </summary>
         /// <param name="siloNames">Only deploy the specified silos which must also be present in the <see cref="Runtime.Configuration.ClusterConfiguration.Overrides"/> collection.</param>
-        public async Task DeployAsync(IEnumerable<string> siloNames)
+        public async Task DeployAsync(IEnumerable<string> siloNames, IServiceProvider serviceProvider)
```

Comment:
Not convinced deploying a service provider is a good idea.  Maybe an IServiceCollection, but a fully populated container sounds like a recipe for badness.